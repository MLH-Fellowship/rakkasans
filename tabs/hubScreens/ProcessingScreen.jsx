import React from 'react';
import {
  View, StyleSheet,
} from 'react-native';
import { useNavigation } from '@react-navigation/native';

import Colors from '../../constants/Colors';
import SquareButton from '../../components/SquareButton';

export default function BattScreen() {
  const navigation = useNavigation();
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: Colors.white,
      width: '100%',
      alignItems: 'center',
      justifyContent: 'center',
    },
    buttonView: {
      width: '100%',
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-evenly',
      paddingBottom: 15,
    },
  });
  return (
    <View style={styles.container}>
      <View style={[styles.buttonView, { paddingTop: 10 }]}>
        <SquareButton
          name="shield"
          text="In"
          buttonSize={90}
          textSize={13}
          onPress={() => navigation.navigate('In Processing')}
        />
        <SquareButton
          name="history"
          text="Out"
          buttonSize={90}
          textSize={13}
          onPress={() => navigation.navigate('Out Processing')}
        />
      </View>
    </View>
  );
}
