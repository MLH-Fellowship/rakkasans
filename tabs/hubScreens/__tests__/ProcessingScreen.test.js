import React from 'react';
import renderer from 'react-test-renderer';

import ProcessingScreen from '../ProcessingScreen';

jest.useFakeTimers();
jest.mock('@react-navigation/native');

it('renders correctly', () => {
  const tree = renderer.create(<ProcessingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
