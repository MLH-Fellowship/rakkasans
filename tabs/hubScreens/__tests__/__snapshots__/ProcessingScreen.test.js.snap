// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 30,
        "paddingBottom": 20,
        "top": "2%",
        "width": 30,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "paddingBottom": 15,
          "width": "100%",
        },
        Object {
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "elevation": 5,
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#a0a0a0",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 90,
              "width": 90,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "paddingBottom": 6,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "andale-mono",
              },
              Object {
                "fontSize": 13,
              },
            ]
          }
        >
          Out
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "elevation": 5,
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#a0a0a0",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 90,
              "width": 90,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "paddingBottom": 6,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "andale-mono",
              },
              Object {
                "fontSize": 13,
              },
            ]
          }
        >
          In
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
</View>
`;
